File: frontend/src/types/order.ts
export const OrderSide = {
    BUY: "BUY",
    SELL: "SELL",
} as const;
export type OrderSide = (typeof OrderSide)[keyof typeof OrderSide];

export const OrderType = {
    LIMIT: "LIMIT",
    MARKET: "MARKET",
} as const;
export type OrderType = (typeof OrderType)[keyof typeof OrderType];

export const OrderStatus = {
    PENDING: "PENDING",
    FILLED: "FILLED",
    CANCELLED: "CANCELLED",
    PARTIALLY_FILLED: "PARTIALLY_FILLED",
    SETTLED: "SETTLED",
} as const;
export type OrderStatus = (typeof OrderStatus)[keyof typeof OrderStatus];

export interface CreateOrderRequest {
    userId: string;
    symbol: string;
    type: OrderType;
    side: OrderSide;
    price: number;
    quantity: number;
}

export interface OrderResponse {
    id: string;
    userId: string;
    symbol: string;
    type: OrderType;
    side: OrderSide;
    price: number;
    quantity: number;
    status: OrderStatus;
    createdAt: string;
}

File: src/main/java/com/trading/modules/order/model/Order.java
package com.trading.modules.order.model;

import java.math.BigDecimal;
import java.util.UUID;

import com.trading.shared.model.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "trading_orders", indexes = {
        @Index(name = "idx_order_user_id", columnList = "user_id"),
        @Index(name = "idx_order_symbol", columnList = "symbol"),
        @Index(name = "idx_order_status", columnList = "status")
})
@Data
@EqualsAndHashCode(callSuper = true)
public class Order extends BaseEntity {

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "symbol", nullable = false, length = 20)
    private String symbol;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false, length = 20)
    private OrderType type;

    @Enumerated(EnumType.STRING)
    @Column(name = "side", nullable = false, length = 10)
    private OrderSide side;

    @Column(name = "price", precision = 19, scale = 8, nullable = false)
    private BigDecimal price;

    @Column(name = "quantity", nullable = false)
    private int quantity;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private OrderStatus status = OrderStatus.PENDING;
}
